concepts/opaque-types/about.md

```diff
@@ -0,0 +1,83 @@
+# About
+
+## Introduction
+
+Opaque Types are an encapsulation technique in Elm. Some people also call this information hiding, or hiding implementation details.
+They are also commonly used to enforce validation, using the [Parse, dont validate][parse-dont-validate] pattern.
+For example, you could have a `EmailAddress` opaque type, that can only contain valid email addresses (as opposed to a `String`, which can contain anything).
+
+## Creating an Opaque Type
+
+[Custom types][custom-types] have one or more *variants*, and the compiler automatically generates a function to create each *variant*, using the *variant* name.
+These are the only way in which custom types can be created.
+The compiler also allows destructuring for these variants.
+
+```elm
+type Maybe a
+    = Nothing --> Variant implies a `Nothing` function, with type: Maybe, and destructuring of 'Nothing'
+    | Just a  --> Variant implies a `Just` function, with type: a -> Maybe, and destructuring of 'Just'
+```
+
+By default, all of this is internal to the file / module, and we use the `exposing` keyword to choose what to export.
+The `Maybe` module exposes the `Maybe` type (so that other modules can use existing `Maybe` values), and also exposes the `Nothing` and `Just` variants (so that other modules can create  and destructure `Maybe` values).
+It uses the following syntax to do this.
+The `Maybe(..)` part tells Elm to expose all the *variants*, as well as the custom type definition. You can also name the *variants* to expose (`Maybe(Just, Nothing)`), but this is less common.
```

Comment:
```suggestion
The `Maybe(..)` part tells Elm to expose all the *variants*, as well as the custom type definition.
You can also name the *variants* to expose (`Maybe(Just, Nothing)`), but this is less common.
```

Reply:
Actually I've just checked, and this used to be the case, but support for it got removed, I think in 0.19.

Reply:
Yep, it was removed because some people used it to only partially expose variants, resulting in user code with match expressions not compiling because of missing cases.